### PR TITLE
Update navigation tabs for code lookup and guide pages

### PR DESCRIPTION
**Pull request by mehansen**
# FRONTEND PULL REQUEST
## Related Issue
#4693 

## Changes Proposed
- change routes for code lookup and guide pages so NavLink tabs are correctly highlighted
  - update links to those pages across other uploader pages
- add back link at top of code lookup page and change card size to match upload guide page
- add a component to scroll to top on mount for the upload guide and code lookup pages
- small accessibility change to the header order on the lookup page that the Deque tool picked up

## Additional Information
the reason it appears that we auto-scroll to the subnav when the code lookup page is that we preserve the scroll location when we navigate between pages in the app. spoke with @lg-king and @vz3 and got feedback that the pages we navigate to from body links should reset scroll location

## Testing
deployed to `dev6` - please check all links work!

## Screenshots / Demos
https://user-images.githubusercontent.com/6401323/203432804-686665bb-7103-468e-8202-0cf97b22c520.mov

### Accessibility
- [x] All changes have run through the Deque automated testing

### Design
- [x] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### CODE REVIEW

File: frontend/src/app/App.tsx [hunk 1]
```diff
@@ -210,44 +210,47 @@ const App = () => {
                 }
               />
               <Route
-                path="csv/codelookup"
+                path="results/upload/submit"
```
Comment (mehansen) [inline frontend/src/app/App.tsx:213]: changed the order we define routes on this page just for readability - submit first, then guide and code-lookup which essentially fall under submit

**Comment by vz3** [comment]
Looks great here can you confirm that the tabbed nav doesn't reset scroll position?

**Comment by mehansen** [comment]
> Looks great here can you confirm that the tabbed nav doesn't reset scroll position?

@vz3 yes it only resets for the Upload guide and Code lookup pages, neither of which can be navigated to from the tab nav

**Review by lg-king — APPROVED** [review]

File: frontend/src/app/commonComponents/ScrollToTopOnMount.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+function ScrollToTopOnMount() {
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, []);
+
+  return null;
+}
+
+export default ScrollToTopOnMount;
```
Comment (mehansen) [inline frontend/src/app/commonComponents/ScrollToTopOnMount.tsx:11]: lifted shamelessly from here: https://v5.reactrouter.com/web/guides/scroll-restoration

File: frontend/src/app/uploads/DeviceLookup/DeviceDetails.tsx [hunk 3]
```diff
@@ -74,7 +74,7 @@ const DeviceDetails = (props: { device: DeviceType }) => {
         <div className="tablet:grid-col">
           <div className="text-field">
             <p>You've selected:</p>
-            <h3>{device.name}</h3>
+            <h2 className="font-sans-md">{device.name}</h2>
```
Comment (mehansen) [inline frontend/src/app/uploads/DeviceLookup/DeviceDetails.tsx:77]: fixes accessibility issue that came up when I ran the axe plugin

File: frontend/src/app/uploads/DeviceLookup/DeviceLookup.tsx [hunk 4]
```diff
@@ -16,12 +16,14 @@ import {
...
 import { useOutsideClick } from "../../utils/hooks";
+import iconSprite from "../../../../node_modules/uswds/dist/img/sprite.svg";
+import { LinkWithQuery } from "../../commonComponents/LinkWithQuery";
+import ScrollToTopOnMount from "../../commonComponents/ScrollToTopOnMount";
 
 import DeviceSearchResults from "./DeviceSearchResults";
 import DeviceDetails from "./DeviceDetails";
 
 interface Props {
-  formTitle: string;
```
Comment (mehansen) [inline frontend/src/app/uploads/DeviceLookup/DeviceLookup.tsx:24]: unused prop

**Review by DanielSass — APPROVED** [review]
⭐

File: frontend/src/app/testResults/uploads/schema.json [hunk 5]
```diff
@@ -249,7 +249,7 @@
                 "SNOMED code from lookup table, example: <em>260373001<em>"
               ],
              "description": [
-               "Use one of the comman values below, or find your device on the <a href=\"/csv/codelookup\" class=\"usa-link\" target=\"_blank\" rel=\"noreferrer noopener\">device code lookup tool</a>.",
+               "Use one of the common values below, or find your device on the <a href=\"/app/results/upload/submit/code-lookup\" class=\"usa-link\" target=\"_blank\" rel=\"noreferrer noopener\">device code lookup tool</a>.",
```
Comment (zdeveloper) [inline frontend/src/app/testResults/uploads/schema.json:252]: what a comman mistake!

**Review by zdeveloper — APPROVED** [review]
lgtm! thank you!

**Pull request merged**